docs/writing-tests/test-runner.md

```diff
@@ -0,0 +1,293 @@
+---
+title: 'Test runner'
+---
+
+Storybook test runner turns all of your stories into executable tests. It is powered by [Jest](https://jestjs.io/) and [Playwright](https://playwright.dev/).
+
+- For those [without a play function](../writing-stories/introduction.md): it verifies whether the story renders without any errors.
+- For those [with a play function](../writing-stories/play-function.md): it also checks for errors in the play function and all assertions passed.
+
+These tests run in a live browser and can be executed via the [command line](#cli-options) or your [CI server](#set-up-ci-to-run-tests).
+
+## Setup
+
+The test runner is a standalone, framework-agnostic addon that runs parallel to your Storybook. You will need to take some additional steps to set it up properly. Detailed below is our recommendation to configure and execute it.
+
+Run the following command to install the addon and the required dependencies.
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'common/storybook-test-runner-install.yarn.js.mdx',
+    'common/storybook-test-runner-install.npm.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+Update your `package.json` scripts and enable the test runner.
+
+```json
+{
+  "scripts": {
+    "test-storybook": "test-storybook"
+  }
+}
+```
+
+Start your Storybook with:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'angular/storybook-run-dev.with-builder.js.mdx',
+    'common/storybook-run-dev.npm.js.mdx',
+    'common/storybook-run-dev.yarn.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+<div class="aside">
+💡 Storybook's test runner requires a running Storybook instance to run all the existing tests.
+</div>
+
+Finally, open a new terminal window and run the test runner with:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'common/storybook-test-runner-execute.yarn.js.mdx',
+    'common/storybook-test-runner-execute.npm.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+## Configure
+
+Test runner offers zero-config support for Storybook. However, you can run `test-storybook --eject` for more fine-grained control. It generates a `test-runner-jest.config.js` file at the root of your project, which you can modify. Additionally, you can extend the generated configuration file and provide [testEnvironmentOptions](https://github.com/playwright-community/jest-playwright#configuration) as the test runner also uses [jest-playwright](https://github.com/playwright-community/jest-playwright) under the hood.
+
+### CLI Options
+
+The test runner is powered by [Jest](https://jestjs.io/) and accepts all its [CLI options](https://jestjs.io/docs/cli) (e.g., `--watch`, `--maxWorkers`).
```

Comment:
```suggestion
The test runner is powered by [Jest](https://jestjs.io/) and accepts all of its [CLI options](https://jestjs.io/docs/cli) (e.g., `--watch`, `--maxWorkers`).
```

Last I knew, @yannbf had purposefully restricted it to only accept a subset of Jest options. That has probably changed and my understanding is mistaken, but calling it out, just in case.

Reply:
I might be mistaken then. @yannbf could you list out which flags are supported?

Reply:
If it is indeed limited, then I think that's likely the table immediately below this paragraph.

Reply:
Based on [this](https://github.com/storybookjs/test-runner/blob/next/src/util/getParsedCliOptions.ts) the test runner CLI is indeed limited to a subset of commands and to avoid confusion the wording here should be instead something like:

"
The test runner is powered by [Jest](https://jestjs.io/) and accepts a subset of its [CLI options](https://jestjs.io/docs/cli) (e.g., `--watch`, `--maxWorkers`).
"

Reply:
Updated this with the wording I've mentioned above in the latest commits. Before marking it as solved I would like to check if that's accurate, because what the Readme lists is not all of them but actually the subset documented.

Reply:
Yup we limited because in case we switch runners we have less trouble to support a subset of options rather than all options. People can still pass whatever option they want if they eject the config file though. Also, if they want to use certain options, they can open an issue and we will support it

Reply:
@yannbf thanks for the follow-up on this! Appreciate it! 🙏